### ISSUE TITLE
Project settings: The upload icon button is always greyed out

### ISSUE THREAD

**Issue by MMK21Hub** [modrinth/code#1640]
### Describe the bug

In the project information section of **Project settings** > **General**, the **upload icon** button is always greyed out, like disabled buttons in other areas of the website. This is despite the button being fully functional.

### Steps to reproduce

1. Open the settings for a project that you own

### Expected behavior

The button's style would match other, similar, non-disabled buttons in other areas of the website.

#### Correct behavior
![A normal "Upload icon" button](https://user-images.githubusercontent.com/50421330/224535245-f5d1e1be-a3a9-43fa-baa0-d078fed2bffe.png)
#### Current behaviour
![A dimmed "Upload icon" button](https://user-images.githubusercontent.com/50421330/224535301-0eaa9f7d-6c0a-47b8-9547-f6f44a1c1f6a.png)

### Code analysis

Here is [the CSS rule that defines styles for disabled buttons](https://github.com/modrinth/knossos/blob/e838fe30defa7d20e82a8f7db8bce8bb5ebd7308/assets/styles/components.scss#L783):
```css
.button-base {
  &:disabled,
  &[disabled] {
    cursor: not-allowed;
    filter: grayscale(50%);
    opacity: 0.5;
    box-shadow: none;
  }
}
```
Note that this rule applies to any buttons with the `disabled` attribute present, regardless of the attribute's value. Unfortunately, this is inconsistent with how the `disabled` attribute functions: while the browser considers an element with `disabled=false` to be enabled, the CSS rule would still apply.

This is indeed the case with the **upload icon** button, which has the attribute `disabled=false`. This is likely caused by [the `:disabled` directive in the source file for project settings](https://github.com/modrinth/knossos/blob/130ece3d2e5e49a1c901c967cc3d40fbcaa3113c/pages/%5Btype%5D/%5Bid%5D/settings/index.vue#L35). However, I'm not sure why Vue sets the attribute to `false` instead of removing it.

In summary, this issue is the result of a discrepancy between the application of the `disabled` attribute and its treatment in the site-wide stylesheet.

### Notes
This issue [was originally reported](https://discord.com/channels/734077874708938864/1083887858487656528/1083887858487656528) on Discord by `@TheKodeToad`, in the `#test-subject-bugs` forum channel. Thanks!
